### PR TITLE
fix(pu): fix data type and deepcopy bug in RND

### PR DESCRIPTION
**Pull request by puyuan1996**
## Description
fix data type and deepcopy bug in rnd.
add a way to set intrinsic_reward_weight automatically and polish rnd annotations.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/box2d/lunarlander/config/lunarlander_rnd_onppo_config.py [hunk 1]
```diff
@@ -12,12 +12,19 @@
...
         learning_rate=5e-4,
         obs_shape=8,
         batch_size=320,
         update_per_collect=4,
+        obs_norm=True,
+        obs_norm_clamp_min=-1,
+        obs_norm_clamp_max=1,
         clear_buffer_per_iters=10,
+        intrinsic_reward_weight=None,
+        # If intrinsic_reward_weight=None, we will automatically set it based on
```
Comment (PaParaZz1) [inline dizoo/box2d/lunarlander/config/lunarlander_rnd_onppo_config.py:25]: add some comments when `intrinsic_reward_weight` is not None

File: ding/reward_model/ngu_reward_model.py [hunk 2]
```diff
@@ -200,10 +200,10 @@ def _compute_intrinsic_reward(
 
         if s > siminarity_max:
             print('s > siminarity_max:', s.max(), s.min())
-            return torch.tensor(0)  # todo
+            return torch.tensor(0)  # NOTE
         if torch.isnan(s):
             print('torch.isnan(s):', s.max(), s.min())
-            return torch.tensor(0)  # todo
+            return torch.tensor(0)  # NOTE
```
Comment (PaParaZz1) [inline ding/reward_model/ngu_reward_model.py:206]: why add `NOTE`

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/box2d/lunarlander/config/lunarlander_rnd_onppo_config.py (hunk 1 above)
Comment (puyuan1996) [reply dizoo/box2d/lunarlander/config/lunarlander_rnd_onppo_config.py:25]: 好的

File: ding/reward_model/ngu_reward_model.py (hunk 2 above)
Comment (puyuan1996) [reply ding/reward_model/ngu_reward_model.py:206]: 可以直接去掉哈，现在应该不会出现torch.isnan的情况

File: ding/reward_model/rnd_reward_model.py [hunk 3]
```diff
@@ -138,14 +134,31 @@ def estimate(self, data: list) -> None:
...
             rnd_reward = rnd_reward.to(data[0]['reward'].device)
             rnd_reward = torch.chunk(rnd_reward, rnd_reward.shape[0], dim=0)
+        """NOTE:
+        Following normalization approach to extrinsic reward seems be not reasonable,
+        because this approach compresses the extrinsic reward magnitude, resulting in less informative reward signals.
+        """
+        # rewards = torch.stack([data[i]['reward'] for i in range(len(data))])
+        # rewards = (rewards - torch.min(rewards)) / (torch.max(rewards) - torch.min(rewards))
+
+        # TODO(pu): how to set intrinsic_reward_weight and intrinsic_reward_rescale automatically?
```
Comment (PaParaZz1) [inline ding/reward_model/rnd_reward_model.py:144]: why are there still this TODO

File: ding/reward_model/rnd_reward_model.py [hunk 4]
```diff
@@ -156,3 +169,14 @@ def collect_data(self, data: list) -> None:
 
     def clear_data(self) -> None:
         self.train_obs.clear()
+
+    def reward_deepcopy(self, train_data):
```
Comment (PaParaZz1) [inline ding/reward_model/rnd_reward_model.py:173]: you can move the default implementation in basic class, and derived class can modify or overwrite it when it needs

Comment (PaParaZz1) [inline ding/reward_model/rnd_reward_model.py:173]: call `reward_deepcopy` method in `estimate` method by default, and remove deepcopy in `serial_pipeline_xxx`

File: ding/entry/serial_entry_reward_model.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
 from ding.reward_model import create_reward_model
 from ding.utils import set_pkg_seed
 from .utils import random_collect
+import copy
```
Comment (PaParaZz1) [inline ding/entry/serial_entry_reward_model.py:16]: remove copy import

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/reward_model/rnd_reward_model.py (hunk 3 above)
Comment (puyuan1996) [reply ding/reward_model/rnd_reward_model.py:144]: still need to manually set intrinsic_reward_rescale now.

File: ding/reward_model/rnd_reward_model.py [hunk 6]
```diff
@@ -100,57 +106,70 @@ def _train(self) -> None:
     def train(self) -> None:
         for _ in range(self.cfg.update_per_collect):
             self._train()
-        # self.clear_data()
 
     def estimate(self, data: list) -> None:
```
Comment (PaParaZz1) [inline ding/reward_model/rnd_reward_model.py:110]: modify return value typing lint

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**